modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2296,39 +2268,24 @@ private void removeWorkflowInstanceFromIndex(long workflowInstanceId, Elasticsea
   /**
    * Update a workflow instance in the API index.
    *
-   * @param workflowInstance
-   *         the workflowInstance to update
-   * @param accessControlList
-   *         the ACL of the event
-   * @param episodeDublincoreCatalog
-   *         the episode dublincore catalog of the event
+   * @param latestWorkflow
+   *         the workflowIndexData to update
    * @param index
    *         the index to update
    */
-  private void updateWorkflowInstanceInIndex(WorkflowInstance workflowInstance, AccessControlList accessControlList,
-          DublinCoreCatalog episodeDublincoreCatalog, ElasticsearchIndex index) {
-    final long workflowInstanceId = workflowInstance.getId();
-    final String eventId = workflowInstance.getMediaPackage().getIdentifier().toString();
+  private void updateWorkflowInstanceInIndex(WorkflowIndexData latestWorkflow, ElasticsearchIndex index) {
+    final long workflowInstanceId = latestWorkflow.getId();
+    final WorkflowState workflowState = WorkflowState.values()[latestWorkflow.getState()];
+    final String eventId = latestWorkflow.getMediaPackageId();
     final String orgId = securityService.getOrganization().getId();
```

Comment:
We removed `runAs(…)`. Therefore, this is no longer correct. Get the organization from `latestWorkflow`.